panos/panorama.py

```diff
@@ -91,6 +91,44 @@ def devicegroup(self):
     def xpath_vsys(self):
         return self.xpath()
 
+    def update_hierarchical_parent(self, parent=None):
+        """Update this device group's hierarchical parent to the specified device group.
+
+        **Modifies the live device**
+
+        This operation results in a job being submitted to the backend, which
+        this function will block until the move is completed.  The return value of
+        this function is what is returned from
+        :meth:`panos.base.PanDevice.syncjob()`.
+
+        Args:
+            parent (str): The device group that should be the parent of this
+                device group in the hierarchy.  A parent of ``None`` means this device
+                group should not have any parents.
+
+        Returns:
+            dict: Job result
+
+        """
+        dev = self.panorama()
+        logger.debug(
+            '{0}: update hierarchical parent for "{1}": {2}'.format(
+                dev.id, self.name, parent
+            )
+        )
+
+        e = ET.Element("request")
+        em = ET.SubElement(e, "move-dg")
+        eme = ET.SubElement(em, "entry")
+        eme.attrib["name"] = self.name
+
+        if parent is not None:
+            ET.SubElement(eme, "new-parent-dg").text = parent
+
+        cmd = ET.tostring(e, encoding="utf-8")
+        resp = dev.op(cmd, cmd_xml=False)
+        return dev.syncjob(resp)
```

Comment:
I don't think we should force an async operation to act synchronously. Perhaps a `sync=False` argument would be better to make this sync optional? Or let the developer call `syncjob()` themselves on the result of this method.